server/api/batch_inventory.py

```diff
@@ -455,6 +473,129 @@ def validate_choice_name_and_get_choice_id(choice_name: str) -> Optional[str]:
+        )
+
+        def validate_choice_name_and_get_choice_id(choice_name: str) -> Optional[str]:
+            choice_id = None
+            for choice in contest.choices:
+                if choice.name == choice_name:
+                    choice_id = choice.id
+                    break
+                # handle capitalization mismatches for the write in column
+                if choice.name.lower() == choice_name.lower() == "write-in":
+                    choice_id = choice.id
+                    break
+
+            if (
+                not choice_id
+                and choice_name
+                and choice_name.lower() != "overvote"
+                and choice_name.lower() != "undervote"
+                # If the user configured a write-in candidate choice when setting up the audit choice_id
+                # will be set in the for loop above. If the audit wasn't configured for write-ins we can parse them out.
+                and choice_name.lower() != "write-in"
+            ):
+                raise UserError(f"Unrecognized choice in CVR file: {choice_name}")
+            return choice_id
+
+        CVR_FILE_ORDER = ["EV.csv", "ED.csv", "MIB1.csv", "MIB2.csv", "Prov.csv"]
+
+        actual_file_names = read_zip_filenames(cvr_file)
+        (file_names, unexpected_files, missing_files) = (
+            diff_file_lists_ignoring_order_and_case(CVR_FILE_ORDER, actual_file_names)
+        )
+
+        logger.info(
+            f"Snapshot ZIP files: expected={CVR_FILE_ORDER}, actual={actual_file_names}, missing={missing_files}, unexpected={unexpected_files}, ordered files to use={file_names}"
+        )
+        if len(unexpected_files) != 0:
+            raise UserError(
+                f"ZIP contains unexpected files: {', '.join(unexpected_files)}"
+            )
+
+        if len(missing_files) != 0:
+            raise UserError(
+                f"ZIP is missing expected files: {', '.join(missing_files)}"
+            )
+
+        cvr_file.seek(0)
+        unzip_files(cvr_file, working_directory)
+        cvr_file.close()
+
+        cvr_file_readers = [
+            csv_reader_for_cvr(retrieve_file(join(working_directory, file_name)))
+            for file_name in file_names
+        ]
+        cvr_csv = read_cvr_snapshots(file_names, cvr_file_readers)
+
+        (_, headers) = next(cvr_csv)
+        header_indices = get_header_indices(headers)
+
+        contest_names = [contest.name for contest in contests]
+        missing_contest_names = set(contest_names) - set(headers)
+        if len(missing_contest_names) != 0:
+            raise UserError(
+                f"CVR file is missing contest names: {', '.join(missing_contest_names)}"
+            )
+
+        for row_index, (source_file_name, row) in enumerate(cvr_csv):
+            batch = column_value(
+                row, "Precinct", row_index + 1, header_indices, required=True
+            )
+            batch_key: BatchKey = (source_file_name, batch)
```

Comment:
Let's remove the `.csv` extension from the source file name when creating the batch key